sycl/test-e2e/Matrix/slm_utils.hpp

```diff
@@ -0,0 +1,163 @@
+//==------------------ slm_utils.hpp  - DPC++ joint_matrix------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifdef OCL
+typedef ushort ushort16ocl __attribute__((ext_vector_type(16)));
+typedef ushort ushort8ocl __attribute__((ext_vector_type(8)));
+typedef ushort ushort4ocl __attribute__((ext_vector_type(4)));
+typedef ushort ushort2ocl __attribute__((ext_vector_type(2)));
+
+extern "C" {
+SYCL_EXTERNAL ushort2ocl __builtin_IB_simd_block_read_2_global_h(
```

Comment:
I think _builtin_IB_ ... are internal interfaces and are not supposed to be used in applications code directly.
Is it possible to replace with something publicly available, like: https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_subgroups.html
?